v1/topdown/object.go

```diff
@@ -21,6 +21,10 @@ func builtinObjectUnion(_ BuiltinContext, operands []*ast.Term, iter func(*ast.T
 		return err
 	}
 
+	if objA.Compare(objB) == 0 {
+		return iter(operands[0])
+	}
```

Comment:
Another cheep check we could do here is if `objA` or `objB` is empty, and then return the other.

Reply:
Great suggestion! I have added that now 👍